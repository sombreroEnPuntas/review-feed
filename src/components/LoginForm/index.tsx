import React, { useState, ChangeEvent, FormEvent } from 'react'
import { Box, Button, Input, Modal } from 'retro-ui'
import styled from 'styled-components'

import Error from '../Error'
import Spacer from '../Spacer'

import useLogin from '../../utils/useLogin'

const Form = styled.form`
  display: flex;
  flex-direction: column;
  align-items: stretch;
`

const LoginForm = () => {
  const [username, setUsername] = useState('')
  const [password, setPassword] = useState('')

  const { error, login, logout, token } = useLogin()

  const handleLogin = async (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    await login(username, password)
  }

  const handleLogout = () => {
    logout()
  }

  return token ? (
    <Button onClick={handleLogout}>{'Logout'}</Button>
  ) : (
    <Modal>
      <Box error={!!error}>
        <Form onSubmit={handleLogin}>
          <Input
            aria-label="username"
            autoComplete="username"
            name="username"
            onChange={(event: ChangeEvent<HTMLInputElement>) =>
              setUsername(event?.target?.value)
            }
            placeholder="username"
            required
            type="text"
            value={username}
          />
          <Spacer />
          <Input
            aria-label="password"
            autoComplete="password"
            name="password"
            onChange={(event: ChangeEvent<HTMLInputElement>) =>
              setPassword(event?.target?.value)
            }
            placeholder="password"
            required
            type="password"
            value={password}
          />
          <Error>{error ? error : '\n\n'}</Error>
          <Button>{`Login`}</Button>
        </Form>
      </Box>
    </Modal>
  )
}

export default LoginForm
