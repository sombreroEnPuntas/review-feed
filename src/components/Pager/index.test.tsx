import { render } from '@testing-library/react'
import React from 'react'
import { ThemeWrapper } from 'retro-ui'

import TestedComponent, { Props } from '.'

// Deps
import LoginForm from '../LoginForm'
import { AccentCTA } from '.'

// Mocks
jest.mock('../LoginForm')

// Mock data
const setMock = () =>
  (LoginForm as jest.Mock).mockImplementation(() => (
    <AccentCTA>{'Logout'}</AccentCTA>
  ))

const getProps = (customProps) => ({
  message: ':v',
  ...customProps,
})

const stageTest = (customProps?: Props) => {
  setMock()

  return render(
    <ThemeWrapper>
      <TestedComponent {...getProps(customProps)} />
    </ThemeWrapper>
  )
}

describe('Pager', () => {
  afterEach(() => {
    // much force-casting, such type-safe, so dark side
    ;(LoginForm as jest.Mock).mockClear()
  })
  afterAll(() => {
    jest.resetAllMocks() // clean .mock
  })

  it(`renders a Pager with a message`, async () => {
    const { container, getByText } = stageTest()

    const appMessage = getByText(/:v/)

    expect(appMessage).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it(`renders a Pager with error`, async () => {
    const testData = {
      error: true,
      message: `418: The resulting entity body MAY be short and stout.`,
    }
    const { container, getByText } = stageTest(testData)

    const appMessage = getByText(
      /418: The resulting entity body MAY be short and stout./
    )

    expect(appMessage).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
