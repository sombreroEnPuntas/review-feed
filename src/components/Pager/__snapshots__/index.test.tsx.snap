// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager renders a Pager with a message 1`] = `
.c0 {
  background-color: #3C3B41;
  padding: 1em;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  color: #ADACB1;
  margin: 0 0 0.5em 0;
  max-width: -webkit-max-content;
  max-width: -moz-max-content;
  max-width: max-content;
  padding: 0;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #D8D7D9;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50px;
}

.c3:before {
  background-color: #92cc41;
  content: '';
  display: inline-block;
  height: 5px;
  width: 20px;
}

.c4 {
  background-color: #D8D7D9;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  background-color: #c9d68f;
  box-shadow: 4px 0 #3C3B41,0 -4px #3C3B41,-4px 0 #3C3B41,0 4px #3C3B41;
  color: #76c442;
  height: 4em;
  margin: 1em;
  overflow: auto;
  padding: 0.25em;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 0.5em;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  background-color: white;
  box-shadow: 4px 0 #4d5256, 0 -4px #4d5256, -4px 0 #4d5256, 0 4px #4d5256;
  color: #212529;
  cursor: pointer;
  display: block;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  margin: 0.5em;
  padding: 0.5em;
  background-color: #ADACB1;
  box-shadow: none;
}

.c9 {
  background-color: white;
  box-shadow: 4px 0 #4d5256, 0 -4px #4d5256, -4px 0 #4d5256, 0 4px #4d5256;
  color: #212529;
  cursor: pointer;
  display: block;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  margin: 0.5em;
  padding: 0.5em;
  background-color: #E4303E;
  box-shadow: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h3
        class="c2"
      >
        review-feed
      </h3>
      <div
        class="c3"
      />
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        :v
      </div>
    </div>
    <div
      class="c6"
    >
      <div
        class="c7"
      >
        <button
          class="sc-AxiKw c8"
        >
          Filter
        </button>
        <button
          class="sc-AxiKw c8"
        >
          Reset
        </button>
      </div>
      <button
        class="sc-AxiKw c9"
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;

exports[`Pager renders a Pager with error 1`] = `
.c0 {
  background-color: #3C3B41;
  padding: 1em;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  color: #ADACB1;
  margin: 0 0 0.5em 0;
  max-width: -webkit-max-content;
  max-width: -moz-max-content;
  max-width: max-content;
  padding: 0;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #D8D7D9;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50px;
}

.c3:before {
  background-color: #e76e55;
  content: '';
  display: inline-block;
  height: 5px;
  width: 20px;
}

.c4 {
  background-color: #D8D7D9;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  background-color: #ffd7cf;
  box-shadow: 4px 0 #3C3B41,0 -4px #3C3B41,-4px 0 #3C3B41,0 4px #3C3B41;
  color: #ce372b;
  height: 4em;
  margin: 1em;
  overflow: auto;
  padding: 0.25em;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 0.5em;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  background-color: white;
  box-shadow: 4px 0 #4d5256, 0 -4px #4d5256, -4px 0 #4d5256, 0 4px #4d5256;
  color: #212529;
  cursor: pointer;
  display: block;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  margin: 0.5em;
  padding: 0.5em;
  background-color: #ADACB1;
  box-shadow: none;
}

.c9 {
  background-color: white;
  box-shadow: 4px 0 #4d5256, 0 -4px #4d5256, -4px 0 #4d5256, 0 4px #4d5256;
  color: #212529;
  cursor: pointer;
  display: block;
  font-family: 'Press Start 2P',monospace;
  font-size: 0.65em;
  font-weight: bold;
  line-height: calc(2 * 1.5em);
  margin: 0.5em;
  padding: 0.5em;
  background-color: #E4303E;
  box-shadow: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h3
        class="c2"
      >
        review-feed
      </h3>
      <div
        class="c3"
      />
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        418: The resulting entity body MAY be short and stout.
      </div>
    </div>
    <div
      class="c6"
    >
      <div
        class="c7"
      >
        <button
          class="sc-AxiKw c8"
        >
          Filter
        </button>
        <button
          class="sc-AxiKw c8"
        >
          Reset
        </button>
      </div>
      <button
        class="sc-AxiKw c9"
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;
